src/replica/replica_stub.h

```diff
@@ -213,6 +213,10 @@ class replica_stub : public serverlet<replica_stub>, public ref_counter
     // TODO: (Tangyanzhao) add some comments
     void on_detect_hotkey(detect_hotkey_rpc rpc);
 
+    // disk rebalaner
```

Comment:
There's no such class called `disk_rebalancer`, (which is also inconsistent with your code filename `src/replica/replica_disk_balancer.cpp`). "rebalance" is inaccurate to this feature, it misguides that this one single operation will touch all replicas on this server. 

I think you should make a big refactoring upon the current implementation.

```
class disk_migrator : public replica_base {
public:
  disk_migrator();

  void on_disk_migrate(replica_disk_migrate_rpc rpc);

private:
  void check_migration_replica_on_disk(const migrate_replica_request &req,
                                       /*out*/ migrate_replica_response &resp);

private:
  replica* _replica;
  task_tracker _tracker;
}
```

Reply:
Yeah, it's better!